### PR TITLE
STRF-11160: [new] Redirects V3, add Import-Export

### PR DESCRIPTION
**Pull request by jordanarldt**
# [DEVDOCS-3978]
https://bigcommercecloud.atlassian.net/browse/DEVDOCS-3978

## What changed?
Add the Redirect Import/Export V3 endpoints to the Redirects swagger documentation.

## Anything else?
https://bigcommercecloud.atlassian.net/browse/PROJECT-3843

[DEVDOCS-3978]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-3978?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/redirects.v3.yml [hunk 1]
```diff
@@ -155,6 +156,215 @@ paths:
...
+        - name: status
+          in: query
+          description: Filters results by the status of the Redirect Import-Export job.
+          style: form
+          explode: false
+          schema:
+            $ref: '#/components/schemas/ImportExportJobStatus'
+        - name: limit
+          in: query
+          description: Determines the number of items returned per page.
```
Comment (slsriehl) [inline reference/redirects.v3.yml:192]: what is the default?

File: reference/redirects.v3.yml [hunk 2]
```diff
@@ -155,6 +156,215 @@ paths:
...
+      parameters:
+        - $ref: '#/components/parameters/ContentTypeFormData'
+      requestBody:
+        content:
+          multipart/form-data:
+            schema:
+              type: object
+              properties:
+                import_file:
+                  description: A CSV file containing a list of Redirects to be imported.
```
Comment (slsriehl) [inline reference/redirects.v3.yml:277]: can you say more about the structure of the CSV file?

File: reference/redirects.v3.yml [hunk 3]
```diff
@@ -155,6 +156,215 @@ paths:
...
+      description: Opens an event stream to receive live updates from an export job.
+      operationId: getRedirectExportEvents
+      responses:
+        '200':
+          description: Stream of export events
+          content:
+            text/event-stream:
+              schema:
+                type: string
+                example: |
```
Comment (slsriehl) [inline reference/redirects.v3.yml:318]: we can probably format this example better -- will look into this when i run my requests.

File: reference/redirects.v3.yml [hunk 4]
```diff
@@ -155,6 +156,215 @@ paths:
...
+      operationId: getRedirectImportEvents
+      responses:
+        '200':
+          description: Stream of import events
+          content:
+            text/event-stream:
+              schema:
+                type: string
+                example: |
+                  event: message
```
Comment (slsriehl) [inline reference/redirects.v3.yml:341]: same note about example formatting.

File: reference/redirects.v3.yml (hunk 1 above)
Comment (jordanarldt) [reply reference/redirects.v3.yml:192]: @slsriehl The default limit is `10`

File: reference/redirects.v3.yml (hunk 2 above)
Comment (jordanarldt) [reply reference/redirects.v3.yml:277]: @slsriehl Sure, the CSV file must have the following headers:
```
Domain,Old Path,New URL/Path,Target Type,Target ID
```

And here's an example:
```
Domain,Old Path,New URL/Path,Target Type,Target ID
your-store.com,/old-path,/new-manual-path,,
your-store.com,/old-product,,Product,12
your-store.com,/old-brand,,Brand,34
your-store.com,/old-category,,Category,56
your-store.com,/old-page,,Page,78
your-store.com,/old-post,,Post,90
```

**Review by slsriehl — APPROVED** [review]

**Pull request merged**